stylix/fonts.nix

```
@@ -8,57 +8,49 @@
 let
   cfg = config.stylix.fonts;
 
-  fontType = lib.types.submodule {
-    options = {
-      package = lib.mkOption {
-        description = "Package providing the font.";
-        type = lib.types.package;
+  mkFontOptions =
+    {
+      fontName,
+      displayName,
+      package,
+    }:
+    {
+      package = lib.mkPackageOption pkgs displayName { default = package; } // {
```

Can simplify further; using `displayName` has no effect because we're shadowing a custom `description` anyway:


```suggestion
      package = lib.mkPackageOption pkgs package { } // {
```